### PR TITLE
Prompt duplicate test added and pipeline removed

### PR DESCRIPTION
**Pull request by Ashtonville**
Added a test for prompt duplicates and changed gen_service to load the model directly instead through a pipeline.

### CODE REVIEW

File: team_red/gen/generation_service.py [hunk 1]
```diff
@@ -12,36 +10,17 @@
...
-            tokenizer = AutoTokenizer.from_pretrained(model)
-            self._pipeline = pipeline(
-                "text-generation",
-                model=model,
-                tokenizer=tokenizer,
-                device_map="auto",
-                return_full_text=False,
-            )
-        return self._pipeline
+    def set_model(self) -> AutoModelForCausalLM:  # type: ignore[no-any-unimported]
```
Comment (aleneum) [inline team_red/gen/generation_service.py:17]: the naming could be improved. a 'setter' method usually gets some parameters to set AND sets something such as a member variable. This method does neither. `def create_model` or `init_model` would be more appropriate.

File: team_red/gen/generation_service.py [hunk 2]
```diff
@@ -56,15 +35,16 @@ def generate(self, parameters: Dict[str, str]) -> GenResponse:
             "\n====== Resolved prompt =====\n\n%s\n\n=============================",
             resolved,
         )
-
-        response = self.pipeline(
+        response = self.set_model()(
```
Comment (aleneum) [inline team_red/gen/generation_service.py:38]: does a model have to be created for each individual `generate` request? That seems to be overkill. Why not initialise a model once as a member variable (in `set_model -> init_model`) and then call `self._model` directly?

Comment (aleneum) [reply team_red/gen/generation_service.py:38]: previously `self.pipeline` used to be a (lazily initialised) getter property. imho that produces easier to comprehend code and is more efficient than initializing a model for each request

Comment (Ashtonville) [reply team_red/gen/generation_service.py]: I changed the way the model is being initialised in the most recent commit.

File: team_red/gen/generation_service.py (hunk 1 above)
Comment (Ashtonville) [reply team_red/gen/generation_service.py]: I agree. Changed it.

**Comment by aleneum** [comment]
I would have used 

```python
@property
def model(self):
    if not self._model:
        self._model = # ...
    return self._model
# def generate ...
    response = self.model(# ...
```

but since the `GenerationService` is lazily initialised anyway (at least in `Bridge`) I don't see an imminent risk of a `self._model` being loaded without being used.

**Review by aleneum — APPROVED** [review]

**Pull request merged**